lib/SPIRV/PreprocessMetadata.cpp

```diff
@@ -94,6 +96,28 @@ bool PreprocessMetadata::runOnModule(Module &Module) {
   return true;
 }
 
+void PreprocessMetadata::preprocessCXXStructorList(
+    SPIRVMDBuilder::NamedMDWrapper &EM, GlobalVariable *V,
+    ExecutionMode EMode) {
+  auto *List = dyn_cast_or_null<ConstantArray>(V->getInitializer());
+  if (!List)
+    return;
+
+  for (Value *V : List->operands()) {
+    auto *Structor = dyn_cast<ConstantStruct>(V);
+    if (!Structor)
+      continue;
+
+    // Each entry in the list is a struct containing 3 members:
+    // (priority, function, data), with function being the entry point.
+    auto *Kernel = dyn_cast<Function>(Structor->getOperand(1));
+    if (!Kernel)
+      continue;
```

Comment:
Can it be converted into an assertion or we expect non-functions in some cases?

Reply:
Likewise, I will change this into a `cast`.